src/app/shared/directives/abstract-form-ui-control.directive.ts

```diff
@@ -0,0 +1,13 @@
+import { Directive, input, model } from '@angular/core';
+import { FormUiControl, ValidationError, WithOptionalField } from '@angular/forms/signals';
+
+@Directive()
+export abstract class AbstractFormUiControlDirective<T> implements FormUiControl {
```

Comment:
@grizlizli, to be in sync with a new Angular naming convention, we should remove `.directive` from the file name and class name as well.